### PR TITLE
"Tags&Search" tests automation MK4

### PR DESCRIPTION
**Pull request by pachlava**
Three more tests automated:
- [x] Tag auto-completes appear when typing in search field
- [x] Typing `tag:` and something else, like `tag:te` results in autocomplete tag results including that something else, e.g. `test`
- [x] Tag suggestions suggest tags regardless of case

### Fix
For this to happen, I needed to create a number of simple methods:
1. `NoteListAssert.tagsSuggestionsNumber(number)` which calls new `NoteList.getTagsSuggestionsNumber()`
2. `NoteListAssert.tagSuggestionExists(tag)` which is extended by new `NoteListAssert.tagSuggestionsExist([tags])`
3. Four one-line methods: `NoteListAssert.tagsSearchHeaderShown()` / `NoteListAssert.tagsSearchHeaderNotShown()` / `NoteListAssert.notesSearchHeaderShown()` / `NoteListAssert.notesSearchHeaderNotShown()` which really call one same new method `NoteListAssert.searchHeaderShown(header, numberOfOccurences)`
4. All methods from above really end up using more generic `Table.getStaticTextsWithExactLabelCount(label)`, `Table.getVisibleNonLabelledCellsNumber()` 
5. Old `Table.getCellsNumber()` was renamed to `Table.getVisibleLabelledCellsNumber()`to better reflect what it does
6. Several new string constants for logging / locating elements.

### Test
1. Running three new tests:
 - `testTagSuggestionsSuggestTagsRegardlessOfCase()`
 - `testTagAutoCompletesAppearWhenTypingInSearchField()`
 - `testTypingTagAndSomethingElseResultsInAutocompleteTagResultsIncludingThatSomethingElse()` 
 
⬆️ Oh my, I don't feel good about the length of the names, but I found no better way to link the code to corresponding tests from [testing-checklist](https://github.com/Automattic/simplenote-ios/blob/develop/TESTING-CHECKLIST.md). Other than that, we'd need to add IDs to the testing checklist, which will add more maintenance. tbh, I have no better ideas atm.

2. Making sure other tests are not affected 

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.

### CODE REVIEW

File: SimplenoteUITests/AssertGeneric.swift [hunk 1]
```diff
@@ -5,6 +5,7 @@ let notExpectedEnding = " is NOT as expected",
     notAbsentEnding = " NOT absent"
 
 let inAllNotesEnding = " in \"All Notes\"",
+    inNoteListEnding = " in \"Note List\"",
```
Comment (mokagio) [inline SimplenoteUITests/AssertGeneric.swift]: Seeing this in an isolated diff made me notice the escaped `"`. 

There's a relatively recent Swift feature called [_Extended Strings Delimiters_](https://docs.swift.org/swift-book/LanguageGuide/StringsAndCharacters.html#ID606) which simplifies this syntax:

```suggestion
    inNoteListEnding = #" in "Note List""#,
```

File: SimplenoteUITests/Generic.swift [hunk 2]
```diff
@@ -41,26 +41,41 @@ func getToAllNotes() {
 
 class Table {
 
-    class func getCellsNumber() -> Int {
-        // We need to count only the table cells that have X = 0
+    class func getVisibleLabelledCellsNumber() -> Int {
+        // We need only the table cells that have X = 0 and a non-empty label
         // otherwise we will include invisible elements from Sidebar pane, when Notes List is open
         // or the elements from Notes List when Settings are open
+        // or the visible tags search suggestrions
```
Comment (mokagio) [inline SimplenoteUITests/Generic.swift]:
```suggestion
        // or the visible tags search suggestions
```

**Review by mokagio — APPROVED** [review]
Looking good. All tests passing on the iPhone 12 Simulator. 👍

File: SimplenoteUITests/AssertGeneric.swift (hunk 1 above)
Comment (pachlava) [reply SimplenoteUITests/AssertGeneric.swift]: Thank you! 🦉

File: SimplenoteUITests/Generic.swift (hunk 2 above)
Comment (pachlava) [reply SimplenoteUITests/Generic.swift]: Thank you!

**Pull request merged**